src/ameba/rule/style/redundant_begin.cr

```diff
@@ -123,7 +123,7 @@ module Ameba::Rule::Style
           break
         when .ident?
           next unless in_body
-          return unless token.value == Crystal::Keyword::BEGIN
+          return unless token.value == :begin
```

Comment:
Using a predicate method is a safer alternative to symbol equality.
```suggestion
          return unless token.value.begin?
```

Reply:
```
fox@xps-13-9370 /t/ameba (master)> shards build
Dependencies are satisfied
Building: ameba
Error target ameba failed to compile:
Showing last frame. Use --error-trace for full trace.

In src/ameba/rule/style/redundant_begin.cr:126:37

 126 | return unless token.value.begin?
                                 ^-----
Error: undefined method 'begin?' for Char (compile-time type is (Char | Crystal::Keyword | String | Nil))
```
Am I doing something wrong or is it not working

Reply:
Symbols are removed as part of this PR https://github.com/crystal-lang/crystal/pull/11871

Reply:
Oh, sry I didn't realize this was a union type. Never mind then. 🙈